lib/validators/application_validator.js

```diff
@@ -114,81 +107,90 @@ function checkForValidValues(jdlApplication) {
     if (optionsToIgnore.includes(option.name)) {
       return;
     }
-    checkForUnknownApplicationOption(option);
-    checkForBooleanValue(option);
-    checkSpecificOptions(option);
+    checkForUnknownApplicationOption(jdlApplication, option);
+    checkForBooleanValue(jdlApplication, option);
+    checkSpecificOptions(jdlApplication, option);
   });
 }
 
-function checkForUnknownApplicationOption(option) {
-  if (!doesOptionExist(option.name)) {
+function checkForUnknownApplicationOption(jdlApplication, option) {
+  if (!jdlApplication.jhipsterOptions.doesOptionExist(option.name)) {
```

Comment:
`jdlApplication.jhipsterOptions.doesOptionExist` should be avoided: what if someone renames the jhipsterOptions attribute from the JDLApp? or removes it? We'd have to change this reference (and others) too.
Having a method in the JDLApplication is better (`jdlApplication.XXXX`). Maybe `jdlApplication.doesJHipsterOptionExist`.

Reply:
I will think about this one.
jhipsterOptions Is an instance of JHipsterOptions that contains all options definitions.
Unless I use JDLApplication directly instead of creating JHipsterOptions.